### PR TITLE
Fix up dependencies and build artifacts for reh-web build

### PR DESCRIPTION
**Pull request by sharon-wang**
### Description

- Addresses: https://github.com/posit-dev/positron/issues/1618
- Updates `build/gulpfile.reh.js` to include the relevant dependencies and build config for reh-web
- Creates `remote/reh-web` which holds a superset of the dependencies between `remote` and `remote/web`
- Adds `remote/reh-web` to the known dirs for dependency installation in `build/npm/dirs.js`
- Adds handling to auto-merge the `package.json` files from `remote` and `remote/web` into `remote-reh-web` in `postinstall.js`
- Fixes up the path to load `react-dom/client.js` from, which is located at the `static` path rather than the `remote` path for a server build

#### Test the package.json auto-merging

1. make a change to `remote/package.json`  and/or `remote/web/package.json`
2. delete the `node_modules` directories in `build` and `remote/reh-web`
3. run `yarn` in the top-level directory
4. no errors should occur and the `node_modules` directories should be recreated, and `remote/reh-web/package.json` should have the changes from `remote/package.json` and `remote/web/package.json` merged into it

#### Testing a release build
See https://github.com/posit-dev/positron-builds/pull/49 for testing steps.

Positron should run in a browser, you should be able to select an interpreter and execute code in the console, create plots, use the Data Explorer, use built-in extensions, etc.

The icon for the Positron tab in in the web browser should be the Positron logo, not the code oss logo.

### CODE REVIEW

**Comment by sharon-wang** [comment]
After pulling in `main`, I'm having trouble getting a local release build working. Resolving that is in progress 🏗️

File: build/gulpfile.vscode.web.js [hunk 1]
```diff
@@ -32,8 +32,10 @@ const WEB_FOLDER = path.join(REPO_ROOT, 'remote', 'web');
 
 const commit = getVersion(REPO_ROOT);
 const quality = product.quality;
+const version = (quality && quality !== 'stable') ? `${packageJson.version}-${quality}` : packageJson.version;
```
Comment (jmcphers) [inline build/gulpfile.vscode.web.js:35]: Does this need a Positron code fence?

**Review by jmcphers — DISMISSED** [review]
LGTM. I already tested the package JSON overlay bits in #4473. For this PR, I built the `reh-web` release and verified that

- the server does not start without a license key
- with a license key, the server starts and boots to a usable Positron environment
image

File: build/gulpfile.vscode.web.js (hunk 1 above)
Comment (sharon-wang) [reply build/gulpfile.vscode.web.js:35]: This line has been restored to the original from upstream https://github.com/microsoft/vscode/blob/7473068e0d7b62bfd4ef4c8f6b86832448ff1d36/build/gulpfile.vscode.web.js#L32, so I think we're good!

But I did miss a fence in the `build/gulpfile.reh.js` file so thank you for bringing this up 🙌

**Pull request merged**